### PR TITLE
Address CodeQL issue from prior PR

### PR DESCRIPTION
**Pull request by batemapf**
## Summary

Addresses Code QL alert from prior [pull](https://github.com/department-of-veterans-affairs/vets-website/pull/25964).

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#0000
- _Link to previous change of the code/bug (if applicable)_
department-of-veterans-affairs/vets-website#0000
- _Link to epic if not included in ticket_
department-of-veterans-affairs/va.gov-team#0000

## Testing done

- _Describe what the old behavior was prior to the change_
- _Describe the steps required to verify your changes are working as expected_
- _Describe the tests completed and the results_
- _Exclusively stating 'Specs and automated tests passing' is NOT acceptable as appropriate testing
- _Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)_

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |       |
| Desktop |        |       |

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/mhv/landing-page/selectors/isLandingPageEnabledForUser.js [hunk 1]
```diff
@@ -19,14 +19,16 @@ import FEATURE_FLAG_NAMES from '~/platform/utilities/feature-toggles/featureFlag
...
-  if (!loggedIn) return false;
   const mhvlpFeatureToggle = FEATURE_FLAG_NAMES.mhvLandingPageEnabled;
   const featureToggleEnabled = toggleValues(state)[mhvlpFeatureToggle];
-  if (!featureToggleEnabled) return false;
   // selectPatientFacilites alters the facilities array using map. You _must_
   // use this selector. DO NOT traverse state. e.g. - state.user.profile.facilities
   const facilities = selectPatientFacilities(state) || [];
   const hasFacilities = facilities.length > 0;
-  return loggedIn && featureToggleEnabled && hasFacilities;
+  if (featureToggleEnabled && loggedIn && hasFacilities) {
```
Comment (carlosfelix2) [inline src/applications/mhv/landing-page/selectors/isLandingPageEnabledForUser.js:30]: Love this reorg with just one return! Thx

**Review by carlosfelix2 — APPROVED** [review]

File: src/applications/mhv/landing-page/selectors/isLandingPageEnabledForUser.js (hunk 1 above)
Comment (batemapf) [reply src/applications/mhv/landing-page/selectors/isLandingPageEnabledForUser.js:30]: thanks! will give it a try. i think i did something like that initially, but the linter complained.

Comment (jzucadi) [reply src/applications/mhv/landing-page/selectors/isLandingPageEnabledForUser.js:30]: perhaps something like `return !!(featureToggleEnabled && loggedIn && hasFacilities);` would work but for readability we might want to stick with your addition.

**Review by jzucadi — COMMENTED** [review]
Looks good, was going to suggest `return` !!(featureToggleEnabled && loggedIn && `hasFacilities);` but for the sake of readability we should stick with your addition.

**Review by jzucadi — APPROVED** [review]

File: src/applications/mhv/landing-page/selectors/isLandingPageEnabledForUser.js (hunk 1 above)
Comment (radavis) [reply src/applications/mhv/landing-page/selectors/isLandingPageEnabledForUser.js:30]: I believe the linter was complaining about the combination of guard clauses (if..return) and a final return statement that included the same conditions. Now that you've removed the guard clauses on lines 23 & 26, you should be able to keep  what was on line 31 without the linter giving you any guff.

**Comment by carlosfelix2** [comment]
@batemapf This PR can be closed as https://github.com/department-of-veterans-affairs/vets-website/pull/26307 replaces that logic anyways.

**Pull request closed without merging**